### ISSUE TITLE
Bug: jqui_sortable doesn't return its order when it is rendered dynamically and there is an orderInput on the page

### ISSUE THREAD

**Issue by daattali** [Yang-Tang/shinyjqui#43]
That's a bit of a complicated title, I know!

Here is an example of what I mean: where there is a `jqui_sortable` and an `orderInput` together in the UI, it is ok:

```
library(shiny)

ui <- fluidPage(
  jqui_sortable(tags$ul(
    id = 'lst',
    tags$li('A'),
    tags$li('B'),
    tags$li('C')
  )),
  orderInput("ls2", "", c("A","B"))
)

server <- function(input, output, session) {
  observe({
    cat(str(input$lst_order))
  })
}

shinyApp(ui, server)
```

As you can see, the order of the sortable elements gets printed to the console.

However, if the `jqui_sortable` is rendered dynamically (either with a `uiOutput`+`renderUI` or with `insertUI()`), and there is an orderInput on the page, then the order of the sortables doesn't get returned.

```
library(shiny)

ui <- fluidPage(
  uiOutput("foo"),
  orderInput("ls2", "", c("A","B"))
)

server <- function(input, output, session) {
  output$foo <- renderUI({
    jqui_sortable(tags$ul(
      id = 'lst',
      tags$li('A'),
      tags$li('B'),
      tags$li('C')
    ))
  })
  observe({
    cat(str(input$lst_order))
  })
}

shinyApp(ui, server)
```

Now the output in the console is NULL and the order isn't saved.

If you remove the orderInput from the UI, then it will work again.

**Comment by Yang-Tang** [comment]
Hi @daattali, sorry for my late reply. I located the cause of bug finally. It should work now.

**Comment by daattali** [comment]
Thanks, I can confirm this works now!

**Comment by lilianchin6** [comment]
Hi! Were you able to get this to work? I'm still getting NULL with the latest version (0.3.2) with the following configuration:

```library(shiny)


ui <- fluidPage(
  uiOutput("foo"),
  verbatimTextOutput("ls1_output"),
  orderInput("ls2", "", c("A","B")),
  verbatimTextOutput("ls2_output")
)

server <- function(input, output, session) {
  output$foo <- renderUI({
    jqui_sortable(tags$ul(
      id = 'ls1',
      tags$li('A'),
      tags$li('B'),
      tags$li('C')
    ))
  })
  
  
  observe({
    cat(str(input$ls1_order))
    cat(str(input$ls2_order))
  })
  
  output$ls1_output <- renderPrint({
    input$ls1_order
  })
  
  output$ls2_output <- renderPrint({
    input$ls2_order
  })
  
}

shinyApp(ui, server)
```

![image](https://user-images.githubusercontent.com/58946130/73383707-d3417a00-4297-11ea-9f01-6d53ffae047e.png)

But, when both orderInput and jqui_sortable are being rendered dynamically through renderUI in server, order is kept tracked of and it works:

```
library(shiny)

# Works when they are both being dynamically updted

ui <- fluidPage(
  uiOutput("foo"),
  verbatimTextOutput("ls1_output"),
  uiOutput("footwo"),
  verbatimTextOutput("ls2_output")
)

server <- function(input, output, session) {
  output$foo <- renderUI({
    jqui_sortable(tags$ul(
      id = 'ls1',
      tags$li('A'),
      tags$li('B'),
      tags$li('C')
    ))
  })
  
  output$footwo <- renderUI({
    orderInput("ls2", "", c("A","B"))
  })
  
  observe({
    cat(str(input$ls1_order))
    cat(str(input$ls2_order))
  })
  
  output$ls1_output <- renderPrint({
    input$ls1_order
  })
  
  output$ls2_output <- renderPrint({
    input$ls2_order
  })
  
}

shinyApp(ui, server)
```

![image](https://user-images.githubusercontent.com/58946130/73384199-cb360a00-4298-11ea-874d-87fe76ac1aea.png)

However, I actually need orderInput and jqui_sortable to be rendered dynamically through a module,  and when it is, I get NULL again, and I'm wondering if my issue is related to this bug at all (please let me know if I should open a separate issue). Example code below:

```
library(shiny)

orderInputUI <- function(id) {
  ns <- NS(id)
  uiOutput(ns("foothree"))
}

orderInputServer <- function(input, output, session) {
  ns <- session$ns
  output$foothree <- renderUI({
    ns <- session$ns
      orderInput(ns("ls3"), "", c("A","B"))
    })
  return(reactive({input[["ls3_order"]]}))
}

ui <- fluidPage(
  uiOutput("foo"),
  verbatimTextOutput("ls1_output"),
  uiOutput("footwo"),
  verbatimTextOutput("ls2_output"),
  orderInputUI("foothree"),
  verbatimTextOutput("ls3_output")#,
  # orderInput("ls4", "", c("A","B")),
  # verbatimTextOutput("ls4_output")
)

server <- function(input, output, session) {
  ls3output <- callModule(orderInputServer, "foothree")
  
  output$foo <- renderUI({
    jqui_sortable(tags$ul(
      id = 'ls1',
      tags$li('A'),
      tags$li('B'),
      tags$li('C')
    ))
  })
  
  output$footwo <- renderUI({
    orderInput("ls2", "", c("A","B"))
  })
  
  
  output$ls3_output <- renderPrint({
    ls3output()
  })
  
  output$ls1_output <- renderPrint({
    input$ls1_order
  })
  
  output$ls2_output <- renderPrint({
    input$ls2_order
  })
  
  # output$ls4_output <- renderPrint({
  #  input$ls4_order
  # })
  
}

shinyApp(ui, server)
```
![image](https://user-images.githubusercontent.com/58946130/73384161-b6f20d00-4298-11ea-8643-6e3340b03d93.png)

To recap:

- Works when both sortable and orderinput are in renderUI in server
- Doesn't work if there's jqui_sortable in server with render UI and an order input in ui directly
- Doesn't work with modules at all (module with orderinput and jqui_sortable in server with render UI)

**Comment by tsolloway** [comment]
@lilianchin6 
I'm having a similar issue. If you use this workaround with version 0.3.2, it'll work in modules and dynamically rendered. Just thought I'd share a short term workaround. 

```r
orderInputHack <- function(inputId, label, items, width = "500px", item_class = "default") {
  div(
    style = paste0("width: ", width),
    tags$strong(label),
    shinyjqui::jqui_sortable(
      div(
        id = inputId,
        style = "font-size: 0;",
        lapply(items, function(item) {
          div(id = item,
              class = paste0("btn btn-", item_class),
              style = "margin: 1px",
              item)
        })
      ),
      options = list(cancel = "")
    )
  )
}
```
